### PR TITLE
feature/streaming-working-on-ecs

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

This PR contains various fixes that make streaming work in ECS

## Changes proposed in this pull request

1. WebSocket scheme between html and django is now `wss` when deployed but `ws` locally
2. WebSocket scheme django and core-api is now hardcoded to be `ws` as they sit on the same network - this is the key change
3. Document extraction for streaming has been fixed
4. Integration tests for streaming have been extended to match non-streaming

## Guidance to review

1. run integration tests
2. deploy and use

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/9345787124

### CODE REVIEW

File: django_app/redbox_app/settings.py [hunk 1]
```diff
@@ -171,7 +171,7 @@
 )
 CSP_STYLE_SRC = ("'self'",)
 CSP_FRAME_ANCESTORS = ("'none'",)
-CSP_CONNECT_SRC = ["'self'"] + [WEBSOCKET_SCHEME + "://" + host for host in HOSTS]
+CSP_CONNECT_SRC = ["'self'", f"ws://{HOST}/ws/chat/", f"wss://{HOST}/ws/chat/"]
```
Comment (gecBurton) [inline django_app/redbox_app/settings.py:174]:
```suggestion
CSP_CONNECT_SRC = ["'self'", f"wss://{HOST}/ws/chat/"]
```

File: django_app/redbox_app/settings.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 SECRET_KEY = env.str("DJANGO_SECRET_KEY")
 ENVIRONMENT = env.str("ENVIRONMENT")
-WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", default="ws")
+WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", "wss")
```
Comment (gecBurton) [inline django_app/redbox_app/settings.py:19]:
```suggestion
WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", "ws")
```

File: django_app/redbox_app/settings.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 SECRET_KEY = env.str("DJANGO_SECRET_KEY")
 ENVIRONMENT = env.str("ENVIRONMENT")
-WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", default="ws")
+WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", "ws")
```
Comment (gecBurton) [inline django_app/redbox_app/settings.py:19]:
```suggestion
WEBSOCKET_SCHEME = env.str("WEBSOCKET_SCHEME", default="ws")
```

File: django_app/redbox_app/redbox_core/consumers.py [hunk 4]
```diff
@@ -30,7 +31,7 @@ async def receive(self, text_data):
             {"role": message.role, "text": message.text} for message in session_messages
         ]
         url = (
-            URL.build(scheme=settings.WEBSOCKET_SCHEME, host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
+            URL.build(scheme="ws", host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/consumers.py:34]: this needs to be hard coded as it is the connection to core-api within the same network so should not be `wss`

File: django_app/redbox_app/redbox_core/views.py [hunk 5]
```diff
@@ -173,7 +173,7 @@ def chats_view(request: HttpRequest, chat_id: uuid.UUID | None = None):
     messages = []
     if chat_id:
         messages = ChatMessage.objects.filter(chat_history__id=chat_id).order_by("created_at")
-    endpoint = URL.build(scheme="ws", host=request.get_host(), path=r"/ws/chat/")
+    endpoint = URL.build(scheme=settings.WEBSOCKET_SCHEME, host=request.get_host(), path=r"/ws/chat/")
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/views.py:176]: this one however is public internet facing and should be `wss` on deployment

File: pyproject.toml [hunk 6]
```diff
@@ -5,7 +5,7 @@ target-version = 'py311'
 [tool.poetry]
 name = "redbox"
 version = "1.0.0"
-description = ""
+description = "RAG application that uses GenAI to chat with and summarise civil service documents"
```
Comment (252afh) [inline pyproject.toml:8]: Civil Service should be capitalised afaik

File: pyproject.toml [hunk 7]
```diff
@@ -66,6 +66,7 @@ ipywidgets="^8.1.2"
 anthropic = "^0.25.8"
 litellm = "^1.37.12"
 openai = "^1.28.0"
+pikepdf = "8.15.1"  # pinned to stop poetry complainning
```
Comment (252afh) [inline pyproject.toml:69]: typo in comment

**Review by 252afh — APPROVED** [review]

File: django_app/redbox_app/redbox_core/consumers.py [hunk 8]
```diff
@@ -29,10 +30,7 @@ async def receive(self, text_data):
         message_history: list[dict[str, str]] = [
             {"role": message.role, "text": message.text} for message in session_messages
         ]
-        url = (
-            URL.build(scheme=settings.WEBSOCKET_SCHEME, host=settings.CORE_API_HOST, port=settings.CORE_API_PORT)
-            / "chat/rag"
-        )
+        url = URL.build(scheme="ws", host=settings.CORE_API_HOST, port=settings.CORE_API_PORT) / "chat/rag"
```
[the remark this answers is not in the document]
Comment (252afh) [reply django_app/redbox_app/redbox_core/consumers.py:33]: ah your comment explaining that was on an outdated version.


> this needs to be hard coded as it is the connection to core-api within the same network so should not be wss

Safe to ignore 👍

**Pull request merged**